src/extension.ts

```diff
@@ -34,9 +35,15 @@ export async function activate(context: ExtensionContext): Promise<void> {
     // Mark the app as already initialized
     store.manifestSelected(manifest)
 
-    const outputChannel = window.createOutputChannel('Flatpak')
-    // Create a Flatpak pty
-    const terminal = new FlatpakTerminal(outputChannel)
+    const terminal = new FlatpakTerminal()
+    const runner = new FlatpakRunner()
+
+    terminal.onDidClose(() => {
```

Comment:
This may create reference cycle, not sure how typescript handles memory

Reply:
I don't think typescript has some way to deal with ref cycles